### ISSUE TITLE
Onetab import not working.

### ISSUE THREAD

**Issue by formlessdao** [josh-berry/tab-stash#149]
### System Details

- **OS Version:** [e.g. Windows 10]
- **Browser Version:** [Firefox 90.0.2]
- **Tab Stash Version:** [ 2.9]

**Installed Browser Extensions:**

- 
- 

### Describe the Bug

I am not able to import a large list of 50000 links from Onetab

**Comment by josh-berry** [comment]
Sorry to hear you're having trouble, but I'm afraid I'm going to need a bit more info to help. Can you please provide the steps you followed to do the import, and a description of exactly what is not working and what symptoms you are seeing?

**Comment by formlessdao** [comment]
> 
> 
> Sorry to hear you're having trouble, but I'm afraid I'm going to need a bit more info to help. Can you please provide the steps you followed to do the import, and a description of exactly what is not working and what symptoms you are seeing?

1. I have a .doc file where I pasted from clipboard the 50000 from Onetab.
2. I clicked the "Import" button available in Tabstash and tried to paste from clipboard after cooping the 50000 links from the .doc file.

**Comment by josh-berry** [comment]
When you say "nothing happens", do you mean the text box isn't getting filled with your links from OneTab at all? Or is the text box getting filled with your pasted links? Note that after you paste, you need to click the blue "Import" button at the bottom to actually do the import.

If you are not able to paste from Word at all (i.e. the import box remains empty), I'm afraid this is something controlled by the browser and not Tab Stash, but here are a couple things you can try:

* Copy/paste in smaller chunks (maybe try with just the first few links just to see if it works, and then if it does, try with larger numbers until you find the limit). It could be that Firefox is having trouble pasting such a large block of data.
* Try pasting in Notepad (or your favorite plain-text editor), then copying from Notepad/the text editor and pasting into Tab Stash. It could be that your Firefox doesn't understand the format that is being copied by Word (although this seems unlikely).

Let me know how it goes!

**Comment by formlessdao** [comment]
"When you say "nothing happens", do you mean the text box isn't getting filled with your links from OneTab at all?"

Yes when I paste those 50000 links nothing is even added.

I tried adding 500 links. These are indeed added but for some reason when I click on the "import" button links start to open one by one as tabs in my browser and I need to cancel them importing because they need to import directly as stashed tabs not open 500 tabs in my browser.

**Comment by josh-berry** [comment]
> Yes, when I paste those 50000 links nothing is even added.

Got it, then this is unfortunately most likely a limitation of Firefox itself.

> I tried adding 500 links. These are indeed added/filled in but for some reason when I click on the "import" button links start to open one by one as tabs in my browser and I need to cancel them importing because they need to import directly as stashed tabs not open 500 tabs in my browser.

This is expected—Tab Stash will open up to four temporary tabs at a time to fetch the title/icon of each link so it can be imported correctly. (There is unfortunately no better way to do this, since the site itself has to be fully loaded to get this info.) When the sites are imported, the tabs will be closed automatically. It will never open all 500 tabs at once (which would obviously be pretty bad for your system resources).

There is a feature request open, #109, to provide an option to change this behavior and import just the URLs (no titles/icons) so that import can proceed faster. You might want to keep an eye on that request if you find this behavior bothersome.

Hope this helps—I'm going to close this out, but feel free to leave more comments if you have any more questions on this!

**Comment by formlessdao** [comment]
Well i already have titles from Onetab. you should install it , add one website then export it and see how it is formatted to have both the title and link(output is a json file with title and corresponding link). There is no reason why you could not make it import links directly from a json file on my PC that Onetab outputs, and not need to be influenced by Firefox's limitations, right?

**Comment by josh-berry** [comment]
> Well i already have titles from Onetab. you should install it , add one website then export it and see how it is formatted to have both the title and link(output is a json file with title and corresponding link).

I actually tested Tab Stash's importer with OneTab back when I first wrote it. OneTab exports a custom text format--it is not JSON. But you are right that it includes titles.

> There is no reason why you could not make it import links directly from a json file on my PC that Onetab outputs, and not need to be influenced by Firefox's limitations, right?

OneTab is just one of many formats the Tab Stash importer tries to support. The way it does this today is by looking only for URLs and ignoring the surrounding text entirely (except for guessing at group titles in some cases). This keeps things very simple and reliable, while still supporting the widest range of options. I'll look into searching for tab titles as part of #109, but I can't promise anything since I don't want to break other formats.
